### ISSUE TITLE
Misleading oneway arrows

### ISSUE THREAD

**Issue by emmexx** [gravitystorm/openstreetmap-carto#4780]
In some roads with 2 oneway ways (one for each direction) at certain zoom levels only one arrow per way is drawn leading to believe that sections of the road are oneway.

Probably should be better to check if there are other "hidden" arrows nearby and hide both rather than displaying only one.

An example here:
[https://www.openstreetmap.org/?mlat=45.5725&mlon=9.1544#map=16/45.5725/9.1544](https://www.openstreetmap.org/?mlat=45.5725&mlon=9.1544#map=16/45.5725/9.1544)
Via Maresciallo Giardino at zoom 16 has 2 arrows, one pointing north at the north edge, the pointing south at the south edge.
At zoom 17 the arrows double and you can visually understand that that is a 2 directions road.
![carto-giardino](https://user-images.githubusercontent.com/1785010/220844064-51f3db62-b981-4350-aeea-2fb550aac7cd.png)

Similar problems with all 2 directions roads that are entered in OSM db as 2 separate ways. 

  maxx

**Comment by imagico** [comment]
Thanks for pointing this out.

This is technically hard because we have in OSM no established way to map a direction separated road (that is usually mapped separately for the two directions) as a distinct entity.  You would have to use relatively fragile and expensive geometric heuristics to decide on which roads to draw a oneway arrow and where not.  That is not meant to say that this is not worth trying.

This is also related to #497, more specifically the problem that our current labeling (and oneway arrows) is strongly affected by the division in which roads are split in mapping.

**Comment by matkoniecz** [comment]
There is https://wiki.openstreetmap.org/wiki/Tag:dual_carriageway%3Dyes but has a minimal use compared to dual carriageways in general.

**Comment by imagico** [comment]
That would be an option - use is very patchy so far though: https://taginfo.openstreetmap.org/tags/dual_carriageway=yes#map

**Comment by matkoniecz** [comment]
Yes, and use is on a tiny tiny tiny part of all carriageways, even if ones that I tagged are no longer large part of all tag uses.
